2996_reflection/reflection.md

```diff
@@ -846,25 +846,25 @@ On Compiler Explorer: [EDG](https://godbolt.org/z/W74qxqnhf), [Clang](https://go
 #include <meta>
 #include <array>
 
-template <typename T, std::size_t N>
-struct struct_of_arrays_impl;
+template <typename T, size_t N>
+constexpr auto struct_of_arrays_impl = [] consteval {
+  struct impl;
 
-consteval auto make_struct_of_arrays(std::meta::info type,
-                                     std::meta::info N) -> std::meta::info {
-  std::vector<std::meta::info> old_members = nonstatic_data_members_of(type);
+  std::vector<std::meta::info> old_members = nonstatic_data_members_of(^^T);
   std::vector<std::meta::info> new_members = {};
   for (std::meta::info member : old_members) {
-    auto type_array = substitute(^^std::array, {type_of(member), N });
-    auto mem_descr = data_member_spec(type_array, {.name = identifier_of(member)});
+    auto array_type = substitute(^^std::array, {
+        type_of(member),
+        std::meta::reflect_value(N),
+    });
+    auto mem_descr = data_member_spec(array_type, {.name = identifier_of(member)});
     new_members.push_back(mem_descr);
   }
-  return define_aggregate(
-    substitute(^^struct_of_arrays_impl, {type, N}),
-    new_members);
-}
+  return std::meta::define_aggregate(^^impl, new_members);
+}();
 
 template <typename T, size_t N>
-using struct_of_arrays = [: make_struct_of_arrays(^^T, ^^N) :];
+using struct_of_arrays = [: struct_of_arrays_impl<T, N> :];
```

Comment:
I think we absolutely need an example or two illustrating what about the original is broken. Also, doesn't this have consequences for linkage? Originally, `struct_of_arrays<T, N>` had external linkage, now doesn't it have... no linkage?

Reply:
^ replaced with better version from the godbolt links that shouldn't have the linkage issue.

I have a list of examples to add; I'll add that to it.